### DISCUSSION TITLE
Troubled by patients missing from Patient List

### DISCUSSION THREAD

**Discussion by artgoldberg** [uwrit/leaf#463] [General]
Hi Folks

While testing deployment of a new clinical domain (procedures) I repeatedly encounter the ["The Leaf server returned demographic data for less patients than were included in the original cohort. This usually indicates that demographic data are missing for these patients."](https://github.com/uwrit/leaf/blob/8300d44767ae0467e8ff4a08b914a2a6bea11430/src/ui-client/src/components/PatientList/RowCount.tsx#L55) warning message.

Comparing output of `sqlcmd` and Leaf executions of the same query, I find the additional patients found by `sqlcmd` 
 equals the count of missing patients in the Patient List warning about dropped patients.

But I don't understand the justification for dropping any patients. In our case the Patient List includes patient ids, which our users will want to have, independent of whether any other data is available for the patients.

This seems very important to me.

Example list of 2 patients is attached.

Arthur

[test_1_leaf_cohort_1634655590941_demographics.csv](https://github.com/uwrit/leaf/files/7375428/test_1_leaf_cohort_1634655590941_demographics.csv)

**Comment by ndobb** [comment]
Hi @artgoldberg, 

Is it possible that any of the patients have the same patient ids? (ie, a collision)

Also, can you confirm that every patient id in `procedure_occurrence` has a corresponding row in `person`?

**Reply by artgoldberg** [reply]
@ndobb 

Good intuition. Much to my surprise, some `patient_id`s in `procedure_occurrence` do not have corresponding rows in `person`.

**Reply by ndobb** [reply]
Phew! (well, I'm sorry to hear that but relieved it's not a bug with Leaf)
